# Week One Coding Task
## j.merritt 
## 2022.09.07

1. Read this dataset into a DataFrame named `hospitals`.
2. Look at the first few rows and then look at the last few. What do you notice?
3. How many rows and columns does this DataFrame have?
4. We don't need all of the columns. Slice the dataframe down so that it only contains the following columns: 'NAME', 'ADDRESS', 'CITY', 'STATE', 'TYPE',  'POPULATION', 'COUNTY', 'COUNTYFIPS', 'NAICS_DESC', 'BEDS'
5. What are the different hospital types (contained in the `TYPE` column)? Which is the most common?
6. Find all hospitals in Nashville, Tennessee. Take a look at the BEDS column. What do you notice? Why do you think this is?
7. Which county in Tennessee has the most hospitals?
8. There are 95 counties in Tennessee. How many of them have no hospitals at all?

## 1
Read this dataset into a DataFrame named `hospitals`.

In [3]:
import pandas as pd

In [4]:
hospitals = pd.read_csv('../data/Hospitals.csv')

## 2
Look at the first few rows and then look at the last few. What do you notice?
A: most entries are in all capital letters. There are 7,596 entires, and multiple columns of extraneous info.

In [7]:
hospitals.head()

X             Y  OBJECTID        ID  \
0 -1.331889e+07  4.346975e+06         1   5793230   
1 -1.322651e+07  4.049626e+06         2  53391362   
2 -1.315620e+07  4.031978e+06         3  11190023   
3 -1.317190e+07  4.041752e+06         4  17090028   
4 -1.313208e+07  4.037270e+06         5  23691706   

                                                NAME  \
0                    CENTRAL VALLEY GENERAL HOSPITAL   
1  LOS ROBLES HOSPITAL & MEDICAL CENTER - EAST CA...   
2                  EAST LOS ANGELES DOCTORS HOSPITAL   
3          SOUTHERN CALIFORNIA HOSPITAL AT HOLLYWOOD   
4                      KINDRED HOSPITAL BALDWIN PARK   

                     ADDRESS             CITY STATE    ZIP           ZIP4  \
0    1025 NORTH DOUTY STREET          HANFORD    CA  93230  NOT AVAILABLE   
1             150 VIA MERIDA  WESTLAKE VILAGE    CA  91362  NOT AVAILABLE   
2    4060 WHITTIER BOULEVARD      LOS ANGELES    CA  90023  NOT AVAILABLE   
3     6245 DE LONGPRE AVENUE        HOLLYWOOD    CA  90028  NOT AVAILABLE   
4  14148 FRANCISQUITO AVENUE     BALDWIN PARK    CA  91706  NOT AVAILABLE   

   ...             VAL_DATE                           WEBSITE       STATE_ID  \
0  ...  2014/02/10 00:00:00      http://www.hanfordhealth.com  NOT AVAILABLE   
1  ...  2014/02/10 00:00:00  http://www.losrobleshospital.com  NOT AVAILABLE   
2  ...  2014/02/10 00:00:00             http://www.elalax.com  NOT AVAILABLE   
3  ...  2014/02/10 00:00:00         http://sch-hollywood.com/  NOT AVAILABLE   
4  ...  2014/02/10 00:00:00      http://www.khbaldwinpark.com  NOT AVAILABLE   

                                    ALT_NAME ST_FIPS        OWNER TTL_STAFF  \
0                              NOT AVAILABLE       6  PROPRIETARY      -999   
1                              NOT AVAILABLE       6  PROPRIETARY      -999   
2                              NOT AVAILABLE       6  PROPRIETARY      -999   
3  HOLLYWOOD COMMUNITY HOSPITAL OF HOLLYWOOD       6  PROPRIETARY      -999   
4                              NOT AVAILABLE       6  PROPRIETARY      -999   

   BEDS         TRAUMA  HELIPAD  
0    49  NOT AVAILABLE        N  
1    62  NOT AVAILABLE        N  
2   127  NOT AVAILABLE        N  
3   100  NOT AVAILABLE        N  
4    95  NOT AVAILABLE        N  

[5 rows x 34 columns]

In [8]:
hospitals.tail()

X             Y  OBJECTID         ID  \
7591 -8.566905e+06  4.728815e+06      7607  183520904   
7592 -1.219624e+07  3.676931e+06      7608  192185607   
7593 -8.084677e+06  5.094112e+06      7609  196706457   
7594 -8.522423e+06  4.718512e+06      7610  191021401   
7595 -1.244921e+07  4.976468e+06      7611  196784108   

                                               NAME  \
7591  ADVENTIST HEALTHCARE WHITE OAK MEDICAL CENTER   
7592      COPPER QUEEN DOUGLAS EMERGENCY DEPARTMENT   
7593                      WHITING FORENSIC HOSPITAL   
7594             J KENT MCNEW FAMILY MEDICAL CENTER   
7595                  UNIVERSITY ORTHOPAEDIC CENTER   

                         ADDRESS            CITY STATE    ZIP           ZIP4  \
7591           11890 HEALING WAY   SILVER SPRING    MD  20904  NOT AVAILABLE   
7592           100 E. 5TH STREET         DOUGLAS    AZ  85607  NOT AVAILABLE   
7593                70 OBRIEN DR      MIDDLETOWN    CT   6457           3945   
7594  175 HARRY S TRUMAN PARKWAY       ANNAPOLIS    MD  21401  NOT AVAILABLE   
7595              590 WAKARA WAY  SALT LAKE CITY    UT  84108  NOT AVAILABLE   

      ...             VAL_DATE  \
7591  ...  2020/05/05 00:00:00   
7592  ...  2020/05/05 00:00:00   
7593  ...  2020/05/08 00:00:00   
7594  ...  2020/05/05 00:00:00   
7595  ...  2020/05/05 00:00:00   

                                                WEBSITE       STATE_ID  \
7591  https://www.adventisthealthcare.com/locations/...  NOT AVAILABLE   
7592  https://cqch.org/cqch-clinics/copper-queen-dou...  NOT AVAILABLE   
7593  https://portal.ct.gov/dmhas/wfh/whiting-forens...  NOT AVAILABLE   
7594  https://aahs.org/locations/j--kent-mcnew-famil...  NOT AVAILABLE   
7595          https://healthcare.utah.edu/orthopaedics/  NOT AVAILABLE   

           ALT_NAME ST_FIPS               OWNER TTL_STAFF  BEDS  \
7591  NOT AVAILABLE      24          NON-PROFIT      -999   178   
7592  NOT AVAILABLE       4          NON-PROFIT      -999  -999   
7593  NOT AVAILABLE       9  GOVERNMENT - STATE      -999   229   
7594  NOT AVAILABLE      24          NON-PROFIT      -999  -999   
7595  NOT AVAILABLE      49  GOVERNMENT - STATE      -999  -999   

             TRAUMA  HELIPAD  
7591  NOT AVAILABLE        N  
7592       LEVEL IV        N  
7593  NOT AVAILABLE        N  
7594  NOT AVAILABLE        N  
7595  NOT AVAILABLE        N  

[5 rows x 34 columns]

In [5]:
#to look at all column names
hospitals.columns

Index(['X', 'Y', 'OBJECTID', 'ID', 'NAME', 'ADDRESS', 'CITY', 'STATE', 'ZIP',
       'ZIP4', 'TELEPHONE', 'TYPE', 'STATUS', 'POPULATION', 'COUNTY',
       'COUNTYFIPS', 'COUNTRY', 'LATITUDE', 'LONGITUDE', 'NAICS_CODE',
       'NAICS_DESC', 'SOURCE', 'SOURCEDATE', 'VAL_METHOD', 'VAL_DATE',
       'WEBSITE', 'STATE_ID', 'ALT_NAME', 'ST_FIPS', 'OWNER', 'TTL_STAFF',
       'BEDS', 'TRAUMA', 'HELIPAD'],
      dtype='object')

## 3 
How many rows and columns does this DataFrame have?
A: 7596 rows, 34 columns

In [10]:
hospitals.shape

(7596, 34)

In [11]:
hospitals.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7596 entries, 0 to 7595
Data columns (total 34 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   X           7596 non-null   float64
 1   Y           7596 non-null   float64
 2   OBJECTID    7596 non-null   int64  
 3   ID          7596 non-null   int64  
 4   NAME        7596 non-null   object 
 5   ADDRESS     7596 non-null   object 
 6   CITY        7596 non-null   object 
 7   STATE       7596 non-null   object 
 8   ZIP         7596 non-null   int64  
 9   ZIP4        7596 non-null   object 
 10  TELEPHONE   7596 non-null   object 
 11  TYPE        7596 non-null   object 
 12  STATUS      7596 non-null   object 
 13  POPULATION  7596 non-null   int64  
 14  COUNTY      7596 non-null   object 
 15  COUNTYFIPS  7596 non-null   object 
 16  COUNTRY     7596 non-null   object 
 17  LATITUDE    7596 non-null   float64
 18  LONGITUDE   7596 non-null   float64
 19  NAICS_CODE  7596 non-null  

# 4
We don't need all of the columns. Slice the dataframe down so that it only contains the following columns: 'NAME', 'ADDRESS', 'CITY', 'STATE', 'TYPE',  'POPULATION', 'COUNTY', 'COUNTYFIPS', 'NAICS_DESC', 'BEDS'

In [6]:
hospitals = hospitals[['NAME', 'ADDRESS', 'CITY', 'STATE', 
                        'TYPE',  'POPULATION', 'COUNTY', 'COUNTYFIPS', 
                        'NAICS_DESC', 'BEDS']]

In [7]:
hospitals.head()

NAME  \
0                    CENTRAL VALLEY GENERAL HOSPITAL   
1  LOS ROBLES HOSPITAL & MEDICAL CENTER - EAST CA...   
2                  EAST LOS ANGELES DOCTORS HOSPITAL   
3          SOUTHERN CALIFORNIA HOSPITAL AT HOLLYWOOD   
4                      KINDRED HOSPITAL BALDWIN PARK   

                     ADDRESS             CITY STATE                TYPE  \
0    1025 NORTH DOUTY STREET          HANFORD    CA  GENERAL ACUTE CARE   
1             150 VIA MERIDA  WESTLAKE VILAGE    CA  GENERAL ACUTE CARE   
2    4060 WHITTIER BOULEVARD      LOS ANGELES    CA  GENERAL ACUTE CARE   
3     6245 DE LONGPRE AVENUE        HOLLYWOOD    CA  GENERAL ACUTE CARE   
4  14148 FRANCISQUITO AVENUE     BALDWIN PARK    CA  GENERAL ACUTE CARE   

   POPULATION       COUNTY COUNTYFIPS                              NAICS_DESC  \
0          49        KINGS      06031  GENERAL MEDICAL AND SURGICAL HOSPITALS   
1          62      VENTURA      06111  GENERAL MEDICAL AND SURGICAL HOSPITALS   
2         127  LOS ANGELES      06037  GENERAL MEDICAL AND SURGICAL HOSPITALS   
3         100  LOS ANGELES      06037  GENERAL MEDICAL AND SURGICAL HOSPITALS   
4          95  LOS ANGELES      06037  GENERAL MEDICAL AND SURGICAL HOSPITALS   

   BEDS  
0    49  
1    62  
2   127  
3   100  
4    95

# 5
What are the different hospital types (contained in the `TYPE` column)? Which is the most common?
A.1:GENERAL ACUTE CARE, CRITICAL ACCESS, PSYCHIATRIC, LONG TERM CARE, REHABILITATION, MILITARY, SPECIAL, CHILDREN, WOMEN, CHRONIC DISEASE
A.2: GENERAL ACUTE CARE is the most common with 4,481 hospitals

In [8]:
hospitals['TYPE'].value_counts()

GENERAL ACUTE CARE    4481
CRITICAL ACCESS       1027
PSYCHIATRIC            759
LONG TERM CARE         431
REHABILITATION         378
MILITARY               216
SPECIAL                145
CHILDREN               130
WOMEN                   20
CHRONIC DISEASE          9
Name: TYPE, dtype: int64

In [ ]:
GENERAL ACUTE CARE    4481
CRITICAL ACCESS       1027
PSYCHIATRIC            759
LONG TERM CARE         431
REHABILITATION         378
MILITARY               216
SPECIAL                145
CHILDREN               130
WOMEN                   20
CHRONIC DISEASE          9
Name: TYPE, dtype: int64

# 6
Find all hospitals in Nashville, Tennessee. Take a look at the BEDS column. What do you notice? Why do you think this is?

A: The women's hospital at centenial has -999 beds. This might be due to missing values in the original data, or a function used to calculate the value from another that was lost in conversion of the original data to a csv.

In [10]:
#don't label new dataframes as previous with a numeric added onto it
hospitals_nashvilleTN = hospitals.loc[(hospitals['CITY'] == 'NASHVILLE') & (hospitals['STATE'] == 'TN')]

In [13]:
#check range of values in a column before further operations to ensure missing values aren't 
#encoded with a large numeric value
hospitals_nashvilleTN
#slice to make more readable view
hospitals_nashvilleTN[['NAME', 'BEDS']]

NAME  BEDS
545                  TRISTAR SKYLINE MEDICAL CENTER   213
3449          SELECT SPECIALTY HOSPITAL - NASHVILLE    70
3485                           CURAHEALTH NASHVILLE    60
5429             THE WOMEN'S HOSPITAL AT CENTENNIAL  -999
5433        METROPOLITAN NASHVILLE GENERAL HOSPITAL   150
6289                  SAINT THOMAS MIDTOWN HOSPITAL   683
6293  VANDERBILT STALLWORTH REHABILITATION HOSPITAL    80
6661           VA MEDICAL CENTER - NASHVILLE CAMPUS   493
6774    SAINT THOMAS HOSPITAL FOR SPECIALTY SURGERY    23
6777                     SAINT THOMAS WEST HOSPITAL   541
6786          TRISTAR SOUTHERN HILLS MEDICAL CENTER   126
6808              TRISTAR CENTENNIAL MEDICAL CENTER   733
6813                 VANDERBILT UNIVERSITY HOSPITAL  1091

# 7
Which county in Tennessee has the most hospitals?

A: Shelby County


In [12]:
#is hospitals.loc necessary here?
(hospitals.loc[hospitals['STATE'] == 'TN'])['COUNTY'].value_counts()

SHELBY        23
DAVIDSON      15
HAMILTON      13
KNOX          11
WASHINGTON     6
              ..
CARTER         1
WARREN         1
CLAY           1
SMITH          1
WAYNE          1
Name: COUNTY, Length: 80, dtype: int64

SHELBY        23
DAVIDSON      15
HAMILTON      13
KNOX          11
WASHINGTON     6
              ..
CARTER         1
WARREN         1
CLAY           1
SMITH          1
WAYNE          1
Name: COUNTY, Length: 80, dtype: int64

# 8 
There are 95 counties in Tennessee. How many of them have no hospitals at all?

A: 15 counties do not have a hospital (95 counties, only 80 listed as having a hospital in this data set, 95 - 80 = 15)

In [26]:
#hospitals_tn_counties = (hospitals.loc[hospitals['STATE'] == 'TN'])['COUNTY'].value_counts()
hospitals_tn = hospitals.loc[hospitals['STATE'] == 'TN']

In [27]:
#hospitals_tn_counties.info()
#alternative
hospitals_tn['COUNTY'].nunique()

80

<class 'pandas.core.series.Series'>
Index: 80 entries, SHELBY to WAYNE
Series name: COUNTY
Non-Null Count  Dtype
--------------  -----
80 non-null     int64
dtypes: int64(1)
memory usage: 1.2+ KB